plonky2x/src/frontend/vars/array.rs

```diff
@@ -162,18 +170,75 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         V::from_variables_unsafe(&selected_vars)
     }
 
-    pub fn get_fixed_subarray<const SIZE: usize>(
+    pub fn get_fixed_subarray<const MAX_ARRAY_SIZE: usize, const SUB_ARRAY_SIZE: usize>(
         &mut self,
-        array: &[Variable],
-        start: Variable,
-    ) -> ArrayVariable<Variable, SIZE> {
-        let mut res = Vec::new();
-        for i in 0..SIZE {
-            let i_variable = self.constant(L::Field::from_canonical_usize(i));
-            let index = self.add(start, i_variable);
-            res.push(self.select_array(array, index));
+        array: &ArrayVariable<Variable, MAX_ARRAY_SIZE>,
+        array_size: Variable,
+        start_idx: Variable,
+        seed: &ArrayVariable<ByteVariable, 15>, // TODO: Seed it with 120 bits.  Need to figure out if this is secure.
+    ) -> ArrayVariable<Variable, SUB_ARRAY_SIZE> {
+        // TODO:  Need to add check that array_size is less than MAX_ARRAY_SIZE.
+        // TODO:  Need to add check that start_idx + SUB_ARRAY_SIZE is less than array_size.
+
+        let sub_array = self.init::<ArrayVariable<Variable, SUB_ARRAY_SIZE>>();
+
+        self.add_simple_generator(SubArrayExtractor::<L, D, MAX_ARRAY_SIZE, SUB_ARRAY_SIZE> {
```

Comment:
I think that we should transition to using the `hint` api everywhere instead of `add_simple_generator`, soi I would change that here